streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -239,7 +246,7 @@ public void run() {
         } catch (Exception e) {
             // we have caught all Kafka related exceptions, and other runtime exceptions
             // should be due to user application errors
-            log.error(String.format("stream-thread [%s] Streams application error during processing: ", this.getName()),  e);
+            log.error(logPrefix + " Streams application error during processing: ",  e);
```

Comment:
I quick question: it seems such optimization is only done in SLF4j 1.6.0+ versions, what if users use another log4j lib or older version of SLF4j?